backend_jobs/ingestion_pipeline/pipeline_lib/image_provider_interface.py

```diff
@@ -0,0 +1,126 @@
+"""
+  Copyright 2020 Google LLC
+  Licensed under the Apache License, Version 2.0 (the "License");
+  you may not use this file except in compliance with the License.
+  You may obtain a copy of the License at
+
+    https://www.apache.org/licenses/LICENSE-2.0
+
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+ """
+ # pylint: disable=pointless-string-statement
+
+from abc import ABC, abstractmethod
+import apache_beam
+
+""" Each ImageProvider that is added to the platform will inherit from the ImageProviders class.
+This class is in charge of:
+* Returning images from the provider.
+* Calculating the number of batches we want to run in parallel.
+* Extracting additional metadata.
+"""
+
+class ImageProvider(ABC, apache_beam.DoFn):
+    """ ImageProvider is an interface that all the image ingestion providers inherit from.
+    """
+    @abstractmethod
+    def __init__(self, query_arguments=None):
+        """ This function is in charge of calling an API/Image provider source
+        and returns a iterable collection of images.
+
+        Args:
+            num_of_page: The number of page we want to retrieve images from.
+        """
+
+    @abstractmethod
+    def get_images(self, num_of_page):
+        """ This function is in charge of calling an API/Image provider source
+        and returns a iterable collection of images.
+
+        Args:
+            num_of_page: The number of page we want to retrieve images from.
+
+        Returns:
+            Iterable collections of objects, can be different between providers.
```

Comment:
can be different between providers -> provider specific

Reply:
changed